src/components/WalletDropdown/MiniPortfolio/Activity/parseLocal.ts

```diff
@@ -182,24 +181,20 @@ export function parseLocalActivity(
   return { ...defaultFields, ...additionalFields }
 }
 
-export function useLocalActivities(): ActivityMap | undefined {
+export function useLocalActivities(account: string): ActivityMap {
   const allTransactions = useMultichainTransactions()
-  const { chainId } = useWeb3React()
   const tokens = useCombinedActiveList()
 
   return useMemo(
     () =>
-      chainId
-        ? allTransactions.reduce((acc: { [hash: string]: Activity }, [transaction, chainId]) => {
-            try {
-              const localActivity = parseLocalActivity(transaction, chainId, tokens)
-              if (localActivity) acc[localActivity.hash] = localActivity
-            } catch (error) {
-              console.error('Failed to parse local activity', transaction)
-            }
-            return acc
-          }, {})
-        : undefined,
-    [allTransactions, chainId, tokens]
+      allTransactions.reduce((acc: { [hash: string]: Activity }, [transaction, chainId]) => {
+        if (transaction.from === account) {
+          const localActivity = parseLocalActivity(transaction, chainId, tokens)
```

Comment:
are you confident that  `parseLocalActivity` can't throw anymore, at all? can it fail in some unexpected way unrelated to the tokens existing in the lists?

Reply:
I can add it back. I wrote it w/ the try/catch as a precaution but now it feels off to me that I call this code "parsing" and use try/catch, when ideally it'd be treated more as "adapting" & since these are all front-end types rely on the type-catcher rather than try/catch.

Or does it make sense to try/catch code that adapts types fetched from localStorage? I.e. in case a redux type is updated and the browser is providing objects w/ invalid fields or something like that? In that case I'm confident keeping the try/catch I just wanna make sure I'm following best practices

Reply:
oh the error here is from looking up currencies that don't exist in our token map (like user imported ones) - we had an incident last week where we fixed the logic on tx notification by handling the `undefined` case but i'm not sure it's being handled here yet

Reply:
@tinaszheng I think @just-toby fixed that by adding ? checks to getCurrency in this file. That fix should prevent errors for anything that uses parseLocalActivity (eddie's tx notification flow and the activity tab on MP). I'm wondering if the try/catch is still important tho for the other reasons I mentioned above

Reply:
ohh i see i see i think you're confident that no other issues can throw then fine to take out 🤔